src/Utilities/Spherepack.hpp

```diff
@@ -0,0 +1,108 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+/// \file
+/// Provides a C++ interface to the Fortran SPHEREPACK library.
+/**************************************************************************
+This is a modified version of spherepack.  The changes are the following:
+
+1) all functions take a stride and an offset argument
+
+2) a subset of the functions will loop over all elements of the stride
+   when offset==-1. Functions supporting this are shags,shsgs,gradgs,vhsgs.
+
+These changes allow 3d-sphere gradient operations, and physical to
+spectral transforms (and vice versa) in a single call into the
+appropriate spherepack functions instead of one per stride.  (Note
+that this currently holds for I1xS2 topology but not for S2xI1
+topology.  Similar optimization can be added, without change of
+spherepack itself, for S2xI1, by making use of its 'nt' argument.).
+
+
+3) shsgs and shags no longer have limitations on the size of the passed-in
+   coefficient arrays 'a' and 'b'.  They also take additional arguments,
+   'mdabmax, 'ndabmax'.  Coefficients that do not fit into the provided
+   storage are ignored / treated as zero / not computed.   The same holds
+   for coefficients with indices above {m,n}dabmax.
+
+This allows for efficient implementation of definite integrals, where
+now once can request compuation of the l=0 coefficient only.  This
+also helps in algorithms that change the number of coefficients (or
+equivalently collocation points), as restriction or prolongation of
+the coefficient array is no longer (explicitly) needed.
+
+
+Given that we so far do not need to call spherepack with symmetries
+enabled, the above changes were not tested for symmetric data.
+In order to avoid surprises, symmetries are currently disabled.  If you
+need symmetries, please test them.
+
+Most modifications made by
+
+Bela Szilagyi
```

Comment:
I'm not sure Bela would want us having him be the contact person for all things SPHEREPACK, I would leave this out.

Reply:
I'm now wondering if this comment should be removed, moving the thing about symmetries into the README. I think the README covers just about all of it, so no point in having the comment twice.